lib/parallel_tests/grouper.rb

```diff
@@ -42,14 +42,23 @@ def add_to_group(group, item, size)
       end
 
       def build_features_with_steps(tests, options)
-        require 'parallel_tests/gherkin/listener'
-        listener = ParallelTests::Gherkin::Listener.new
-        listener.ignore_tag_pattern = Regexp.compile(options[:ignore_tag_pattern]) if options[:ignore_tag_pattern]
-        parser = ::Gherkin::Parser::Parser.new(listener, true, 'root')
+        require 'gherkin/parser'
+        ignore_tag_pattern = options[:ignore_tag_pattern].nil? ? nil : Regexp.compile(options[:ignore_tag_pattern])
+        parser = ::Gherkin::Parser.new
+        # format of hash will be FILENAME => NUM_STEPS
+        steps_per_file = {}
         tests.each do |file|
```

Comment:
can use `.each_with_object` ... ideally simply use .map since this returns an array a the end anyway